MultiServer.py

```diff
@@ -495,10 +505,20 @@ def set_save(self, savedata: dict):
 
     # rest
 
+    def calculate_hint_costs(self):
+        self.player_hint_costs = {}
+        for player in self.games.keys():
+            if self.hint_cost and player in self.locations:
+                self.player_hint_costs[player] = max(0, int(self.hint_cost * 0.01 * len([loc for loc in self.locations[player].values()])))
+            else:
+                self.player_hint_costs[player] = 0
+
     def get_hint_cost(self, slot):
-        if self.hint_cost:
-            return max(0, int(self.hint_cost * 0.01 * len(self.locations[slot])))
-        return 0
+        try:
```

Comment:
if you run calculate_hint_costs on multidata load and sasvegame load you should never have a KeyError, correct?